### PR TITLE
TransactionView: inline simple accessors

### PR DESCRIPTION
**Pull request by apfitzge**
#### Problem
- Many simple accessors are not inlined so they are extremely slow

#### Summary of Changes
- Inline them

Fixes #

### CODE REVIEW

**Comment by apfitzge** [comment]
<details>
<summary>Benchmark Results (compared to master)</summary>

```
min sized transactions/TransactionView (Sanitized)
                        time:   [5.6989 µs 5.7085 µs 5.7199 µs]
                        thrpt:  [179.02 Melem/s 179.38 Melem/s 179.69 Melem/s]
                 change:
                        time:   [-54.382% -54.220% -54.038%] (p = 0.00 < 0.05)
                        thrpt:  [+117.57% +118.44% +119.21%]
                        Performance has improved.

simple transfers/TransactionView (Sanitized)
                        time:   [8.8176 µs 8.8353 µs 8.8536 µs]
                        thrpt:  [115.66 Melem/s 115.90 Melem/s 116.13 Melem/s]
                 change:
                        time:   [-37.441% -37.185% -36.903%] (p = 0.00 < 0.05)
                        thrpt:  [+58.487% +59.197% +59.849%]
                        Performance has improved.

packed transfers/TransactionView (Sanitized)
                        time:   [481.01 µs 482.36 µs 483.77 µs]
                        thrpt:  [2.1167 Melem/s 2.1229 Melem/s 2.1289 Melem/s]
                 change:
                        time:   [-16.266% -15.967% -15.667%] (p = 0.00 < 0.05)
                        thrpt:  [+18.578% +19.000% +19.426%]
                        Performance has improved.

packed noops/TransactionView (Sanitized)
                        time:   [2.7080 ms 2.7103 ms 2.7132 ms]
                        thrpt:  [377.41 Kelem/s 377.82 Kelem/s 378.14 Kelem/s]
                 change:
                        time:   [-18.850% -18.635% -18.450%] (p = 0.00 < 0.05)
                        thrpt:  [+22.624% +22.904% +23.229%]
                        Performance has improved.

packed atls/TransactionView (Sanitized)
                        time:   [243.50 µs 243.93 µs 244.46 µs]
                        thrpt:  [4.1888 Melem/s 4.1979 Melem/s 4.2053 Melem/s]
                 change:
                        time:   [-17.666% -17.195% -16.717%] (p = 0.00 < 0.05)
                        thrpt:  [+20.073% +20.766% +21.456%]
                        Performance has improved.
```

</details>

**Comment by apfitzge** [comment]
<details>
<summary>Benchmark Results (compare to other deserialization with and without sanitizization) </summary>

```
min sized transactions/VersionedTransaction
                        time:   [224.69 µs 226.02 µs 227.34 µs]
                        thrpt:  [4.5042 Melem/s 4.5306 Melem/s 4.5573 Melem/s]
min sized transactions/SanitizedVersionedTransaction
                        time:   [242.29 µs 243.13 µs 244.05 µs]
                        thrpt:  [4.1958 Melem/s 4.2117 Melem/s 4.2263 Melem/s]
min sized transactions/TransactionView
                        time:   [5.2354 µs 5.2600 µs 5.2857 µs]
                        thrpt:  [193.73 Melem/s 194.68 Melem/s 195.59 Melem/s]
min sized transactions/TransactionView (Sanitized)
                        time:   [6.4548 µs 6.4639 µs 6.4741 µs]
                        thrpt:  [158.17 Melem/s 158.42 Melem/s 158.64 Melem/s]

simple transfers/VersionedTransaction
                        time:   [476.41 µs 479.83 µs 483.50 µs]
                        thrpt:  [2.1179 Melem/s 2.1341 Melem/s 2.1494 Melem/s]
simple transfers/SanitizedVersionedTransaction
                        time:   [492.76 µs 494.58 µs 496.27 µs]
                        thrpt:  [2.0634 Melem/s 2.0704 Melem/s 2.0781 Melem/s]
simple transfers/TransactionView
                        time:   [6.4250 µs 6.4420 µs 6.4602 µs]
                        thrpt:  [158.51 Melem/s 158.96 Melem/s 159.38 Melem/s]
simple transfers/TransactionView (Sanitized)
                        time:   [8.5108 µs 8.5479 µs 8.5905 µs]
                        thrpt:  [119.20 Melem/s 119.80 Melem/s 120.32 Melem/s]

packed transfers/VersionedTransaction
                        time:   [6.3370 ms 6.3461 ms 6.3578 ms]
                        thrpt:  [161.06 Kelem/s 161.36 Kelem/s 161.59 Kelem/s]
packed transfers/SanitizedVersionedTransaction
                        time:   [6.3877 ms 6.3936 ms 6.4008 ms]
                        thrpt:  [159.98 Kelem/s 160.16 Kelem/s 160.31 Kelem/s]
packed transfers/TransactionView
                        time:   [238.86 µs 239.29 µs 239.68 µs]
                        thrpt:  [4.2723 Melem/s 4.2793 Melem/s 4.2871 Melem/s]
packed transfers/TransactionView (Sanitized)
                        time:   [483.79 µs 484.25 µs 484.76 µs]
                        thrpt:  [2.1124 Melem/s 2.1146 Melem/s 2.1166 Melem/s]

packed noops/VersionedTransaction
                        time:   [4.7106 ms 4.7134 ms 4.7168 ms]
                        thrpt:  [217.10 Kelem/s 217.25 Kelem/s 217.38 Kelem/s]
packed noops/SanitizedVersionedTransaction
                        time:   [4.9270 ms 4.9306 ms 4.9352 ms]
                        thrpt:  [207.49 Kelem/s 207.68 Kelem/s 207.84 Kelem/s]
packed noops/TransactionView
                        time:   [1.3844 ms 1.3886 ms 1.3934 ms]
                        thrpt:  [734.88 Kelem/s 737.41 Kelem/s 739.65 Kelem/s]
packed noops/TransactionView (Sanitized)
                        time:   [2.8322 ms 2.8412 ms 2.8505 ms]
                        thrpt:  [359.24 Kelem/s 360.42 Kelem/s 361.56 Kelem/s]

packed atls/VersionedTransaction
                        time:   [3.4939 ms 3.4958 ms 3.4981 ms]
                        thrpt:  [292.73 Kelem/s 292.93 Kelem/s 293.08 Kelem/s]
packed atls/SanitizedVersionedTransaction
                        time:   [3.6087 ms 3.6185 ms 3.6285 ms]
                        thrpt:  [282.21 Kelem/s 282.99 Kelem/s 283.76 Kelem/s]
packed atls/TransactionView
                        time:   [125.67 µs 125.81 µs 125.94 µs]
                        thrpt:  [8.1307 Melem/s 8.1394 Melem/s 8.1482 Melem/s]
packed atls/TransactionView (Sanitized)
                        time:   [250.19 µs 251.36 µs 252.47 µs]
                        thrpt:  [4.0559 Melem/s 4.0739 Melem/s 4.0929 Melem/s]
```

</details>

Results here show that sanitization is expensive.
Believe most of this additional time is coming from the repeated iteration over instructions and ATLs.

For legacy transactions, we could actually avoid repeating the iteration over instructions by doing the program_id and account checks in the initial loop.
For v0's we do not know the total number of accounts because we haven't parsed the ATLs yet, so cannot do that.
We cannot parse ATLs without first parsing instructions because they are not fixed size so we have no idea where ATLs begins 🫠

**Comment by apfitzge** [comment]
^ As mentioned above a quick demo of doing "inline" sanitization - we could improve perf of the packed transfer and packed noops by roughly 100%.
But all that gain could be simply mitigated by a malicious sender since all that's required to bypass it is attaching a v0 byte 😢 

It also adds complexity since sanitization is done is different places. So opting to not do that for now.

**Review by tao-stones — APPROVED** [review]
lgtm, thanks for sharing bench results.

File: transaction-view/src/instructions_meta.rs [hunk 1]
```diff
@@ -81,6 +81,7 @@ pub struct InstructionsIterator<'a> {
 impl<'a> Iterator for InstructionsIterator<'a> {
     type Item = SVMInstruction<'a>;
 
+    #[inline]
```
Comment (jstarry) [inline transaction-view/src/instructions_meta.rs:84]: is there a tradeoff to inlining a fairly large function like this?

Comment (apfitzge) [reply transaction-view/src/instructions_meta.rs:84]: probably some compile time cost, binary size - but for 20% performance increase (most of it came from this inlining on the "challenging" tx benchmarks) it seems almost certainly worth it to me.

This also doesn't require inlining, just suggests it and lets code outside this crate inline it as well.

**Pull request merged**